compiler/rustc_hir_analysis/src/coherence/inherent_impls.rs

```diff
@@ -186,7 +186,11 @@ impl<'tcx> InherentCollect<'tcx> {
             return;
         };
 
-        let self_ty = self.tcx.type_of(item.owner_id);
+        let mut self_ty = self.tcx.type_of(item.owner_id);
+        if matches!(self_ty.kind(), ty::Alias(..) | ty::Param(_)) {
```

Comment:
Why is `Param` included here? We don't need to normalize parameter types. Also, this should probably only consider projections:

```suggestion
        if matches!(self_ty.kind(), ty::Alias(ty::Projection, ..)) {
```

Reply:
Arf, sorry I forgot to remove it. I copied/pasted the match branch and forgot to clean up.